src/App.js

```diff
@@ -1,26 +1,46 @@
 import {nanoid} from 'nanoid';
-import {useState, useRef} from 'react';
+import {useState, useEffect, useRef} from 'react';
 import styled from 'styled-components';
 
 import {mazeArray} from './arrays/mazeArray.js';
-import MagicForm from './components/MagicForm/MagicForm';
+import MagicForm from './components/MagicForm/MagicForm.js';
+import MagicImage from './components/MagicImage/MagicImage.js';
 import MagicList from './components/MagicList/MagicList.js';
-import {CodeSpan} from './components/Stylesheet/StyledSpans.js';
 import executeSpell from './functionsfolder/executeSpell.js';
 import findCommandMessage from './functionsfolder/findCommandMessage.js';
+import mazeEnd from './images/mazeRooms/mazeEnd.png';
+import mazeWay from './images/mazeRooms/mazeWay.png';
+import {BackgroundImg} from './stylesheet/StyledImages.js';
+import {CodeSpan} from './stylesheet/StyledSpans.js';
 
 export default function App() {
   const [currentArrayPosition, setCurrentArrayPosition] = useState(mazeArray[0]); //this holds a {object}
 
   function changePosition(newPositionAsString) {
     const newPositionAsObject = mazeArray.find(mazeRoom => mazeRoom.path === newPositionAsString);
     setCurrentArrayPosition(newPositionAsObject);
+    setIsRoomEnlighten(false);
   }
 
   const [savedOrders, setSavedOrders] = useState([]);
 
   const ref = useRef();
 
+  const [isRoomEnlighten, setIsRoomEnlighten] = useState(false);
+  function changeEnlighten() {
+    setIsRoomEnlighten(true);
+  }
+  useEffect(() => {}, [isRoomEnlighten]); //used to rerender idependly from useEffect below
+
+  const [currentBackgroundImage, setCurrentBackgroundImage] = useState();
+  useEffect(() => {
+    if (currentArrayPosition.type === 'start' || currentArrayPosition.type === 'end') {
+      setCurrentBackgroundImage(<BackgroundImg src={mazeEnd} alt="the end of the way" />);
+    } else {
+      setCurrentBackgroundImage(<BackgroundImg src={mazeWay} alt="a lonely and spooky corridor" />);
+    }
```

Comment:
I would probably avoid setting an image directly as state, but rather just the path as a string - or an object with `path` and `altText` properties. That way you can directly add your `BackgroundImg` component to the `MagicImage` component and don't have to repeat yourself here.
```suggestion
    if (currentArrayPosition.type === 'start' || currentArrayPosition.type === 'end') {
      setCurrentBackgroundImage({path: mazeEnd, altText: "the end of the way"});
    } else {
      setCurrentBackgroundImage({path: mazeWay, altText: "a lonely and spooky corridor"});
    }
```
If you do that, I would also add an empty object as initial state.

Reply:
ok, this looks waay more elegant, but it isnt working :(

Reply:
What exactly doesn't work though? :)